libs/payments/stripe/src/lib/stripe.service.ts

```diff
@@ -24,38 +20,42 @@ export class StripeService {
     subscriptionId: string,
     promotionId: string
   ) {
-    try {
-      const subscription = await this.stripeManager.retrieveSubscription(
-        subscriptionId
+    const subscription = await this.stripeManager.retrieveSubscription(
+      subscriptionId
+    );
+    if (subscription?.status !== 'active')
+      throw new PromotionCodeCouldNotBeAttachedError(
+        'Subscription is not active'
       );
```

Comment:
This is where the data around the method is attached, subscription id, promotion code id, etc.